pulsar-metadata/src/main/java/org/apache/pulsar/metadata/cache/impl/MetadataCacheImpl.java

```diff
@@ -152,10 +155,14 @@ public CompletableFuture<T> readModifyUpdateOrCreate(String path, Function<Optio
                         try {
                             // Use clone and CAS zk to ensure thread safety
                             clone = serde.deserialize(path, serde.serialize(path, entry.getValue()), entry.getStat());
+                            if (clone == null) {
+                                currentValue = Optional.empty();
+                            } else {
+                                currentValue = Optional.of(clone);
+                            }
                         } catch (IOException e) {
                             return FutureUtils.exception(e);
                         }
-                        currentValue = Optional.of(clone);
```

Comment:
instead of making the change above, `Optional.of` could be changed to `Optional.ofNullable` to achieve the same outcome.